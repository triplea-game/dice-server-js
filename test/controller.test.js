jest.mock('liquidjs', () => () => ({
  name: 'Liquid Module',
  args: arguments,
  express: jest.fn().mockReturnValue('Express Router'),
}));
jest.mock('../src/api/api', () => () => 'API Module');
jest.mock('../src/user/user', () => () => 'User Module');

const bodyParser = require('body-parser');
bodyParser.json = jest.fn(() => 'JSON Parser');
bodyParser.urlencoded = jest.fn().mockReturnValue('urlencoded');

const mockedApp = {
  use: jest.fn(),
  engine: jest.fn(),
  set: jest.fn(),
  listen: jest.fn(),
};

const mockRouter = {
  use: jest.fn(),
};
const mockExpress = jest.fn().mockReturnValue(mockedApp);
mockExpress.Router = jest.fn().mockReturnValue(mockRouter);
mockExpress.static = jest.fn().mockReturnValue('Static');

jest.mock('express', () => mockExpress);

const controller = require('../src/controller');

describe('The controller', () => {
  it('should setup the routes correctly', () => {
    controller(1337, {});

    expect(mockExpress).toMatchSnapshot();
    expect(mockExpress.Router).toMatchSnapshot();
    expect(mockRouter.use).toMatchSnapshot();
    expect(mockedApp.use).toMatchSnapshot();
    expect(mockedApp.engine).toMatchSnapshot();
    expect(mockedApp.set).toMatchSnapshot();
    expect(mockedApp.listen).toMatchSnapshot();
  });
});
