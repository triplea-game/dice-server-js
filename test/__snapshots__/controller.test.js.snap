// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The controller should setup the routes correctly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Object {
        "engine": [MockFunction] {
          "calls": Array [
            Array [
              "html",
              "Express Router",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "listen": [MockFunction] {
          "calls": Array [
            Array [
              1337,
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "set": [MockFunction] {
          "calls": Array [
            Array [
              "views",
              Array [
                "./public/partials",
                "./public/views",
              ],
            ],
            Array [
              "view engine",
              "html",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "use": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
            Array [
              [Function],
            ],
            Array [
              "Static",
            ],
            Array [
              Object {
                "use": [MockFunction] {
                  "calls": Array [
                    Array [
                      "/api",
                      "API Module",
                    ],
                    Array [
                      "/",
                      "User Module",
                    ],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                },
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      },
    },
  ],
}
`;

exports[`The controller should setup the routes correctly 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "caseSensitive": true,
        "strict": true,
      },
    ],
    Array [
      Object {
        "caseSensitive": true,
        "strict": true,
      },
    ],
    Array [
      Object {
        "caseSensitive": true,
        "strict": true,
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Object {
        "use": [MockFunction] {
          "calls": Array [
            Array [
              "/api",
              "API Module",
            ],
            Array [
              "/",
              "User Module",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      },
    },
    Object {
      "type": "return",
      "value": Object {
        "use": [MockFunction] {
          "calls": Array [
            Array [
              "/api",
              "API Module",
            ],
            Array [
              "/",
              "User Module",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      },
    },
    Object {
      "type": "return",
      "value": Object {
        "use": [MockFunction] {
          "calls": Array [
            Array [
              "/api",
              "API Module",
            ],
            Array [
              "/",
              "User Module",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      },
    },
  ],
}
`;

exports[`The controller should setup the routes correctly 3`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/api",
      "API Module",
    ],
    Array [
      "/",
      "User Module",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`The controller should setup the routes correctly 4`] = `
[MockFunction] {
  "calls": Array [
    Array [
      [Function],
    ],
    Array [
      [Function],
    ],
    Array [
      "Static",
    ],
    Array [
      Object {
        "use": [MockFunction] {
          "calls": Array [
            Array [
              "/api",
              "API Module",
            ],
            Array [
              "/",
              "User Module",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`The controller should setup the routes correctly 5`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "html",
      "Express Router",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`The controller should setup the routes correctly 6`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "views",
      Array [
        "./public/partials",
        "./public/views",
      ],
    ],
    Array [
      "view engine",
      "html",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`The controller should setup the routes correctly 7`] = `
[MockFunction] {
  "calls": Array [
    Array [
      1337,
      [Function],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
